sources/zafira-client/src/main/java/com/qaprosoft/zafira/listener/ZafiraListener.java

```diff
@@ -467,14 +467,17 @@ public void onFinish(ITestContext context) {
     @Override
     public void run(IHookCallBack hookCallBack, ITestResult testResult) {
         if (!ZAFIRA_ENABLED) {
+            LOGGER.info("IHookCallBack: zafira not connected so running the test body");
```

Comment:
what "IHookCallBack" does mean?

Reply:
"TestNG allows you to override and possibly skip the invocation of test methods. One example of where this is useful is if you need to your test methods with a specific security manager. You achieve this by providing a listener that implements IHookable."
https://testng.org/doc/documentation-main.html#ihookable